filebeat/input/v2/input.go

```diff
@@ -86,9 +87,37 @@ type Context struct {
 	// Agent provides additional Beat info like instance ID or beat name.
 	Agent beat.Info
 
+	// MetricsRegistry to collect metrics for this input.
+	//
+	// This registry is created on the beat.Info.Monitoring.Namespace. For
+	// compatibility, beats are configured with the global 'dataset' namespace
+	// in beat.Info.Monitoring.Namespace. When running as an OTel receiver, each
+	// beat.Beat instance uses a unique metrics namespace.
+	//
+	// If the input has no ID, this is a 'discard' registry associated to no
+	// namespace.
+	// The inputs still need to call inputmon.NewInputRegistry to further
+	// populate the registry by adding `id` and `input` (the input type) so the
+	// metrics will be published on the `/inputs/` monitoring endpoint.
+	// This allows the inputs to decide to publish or not metrics, regardless of
+	// the input having an ID. A valid metrics registry which will be passed
+	// down to the pipeline client and output. Hoerver if the input does not
+	// call inputmon.NewInputRegistry for the same ID to further populate the
+	// Also, calling inputmon.NewInputRegistry allows inputs to set their type,
+	// the `input` field, as they will. Not all inputs use their hard-coded
+	// name as their type, thus the input itself needs to set it.
```

Comment:
It seems some of the lines are out of place, see L105-106:
```
	// call inputmon.NewInputRegistry for the same ID to further populate the
	// Also, calling inputmon.NewInputRegistry allows inputs to set their type,
```

Could you review/clarify this comment?